src/app/vm/vm-creation/containers/vm-creation.container.ts

```diff
@@ -16,7 +16,7 @@ import { AuthService } from '../../../shared/services/auth.service';
 import { BaseTemplateModel } from '../../../template/shared';
 import { VmService } from '../../shared/vm.service';
 import { NotSelected, VmCreationState } from '../data/vm-creation-state';
-import { KeyboardLayout } from '../keyboards/keyboards.component';
+import { KeyboardLayout } from '../../../settings/components/keyboards/keyboards.component';
```

Comment:
Do we need this type in this file and function that it uses?

Reply:
@tamazlykar we use this type in a keyboard.component too.
where should we store it?

Reply:
As I know you need to remove a keyboard layout from VM creation form.
This is a VM creation form and this type used only in `public onKeyboardChange(keyboard: KeyboardLayout)` do we still need this function and corresponding event emitter and so on?

If somwere else you need this type and it used by different modules it will be better to place type in app/shared/types